.github/workflows/test_with_clone.yaml

```diff
@@ -0,0 +1,189 @@
+                echo "role = \"${SNOWFLAKE_CLONE_ROLE}\"" >> $config_file
+                echo "password = \"${SNOWFLAKE_PASSWORD}\"" >> $config_file
+                echo "warehouse = \"${SNOWFLAKE_WAREHOUSE}\"" >> $config_file
+                echo 'authenticator = "SNOWFLAKE"' >> $config_file
+
+                # Config file for *ADMIN
+                echo '[connections.admin]' >> $config_file
+                echo "account = \"${SNOWFLAKE_ACCOUNT}\"" >> $config_file
+                echo "user = \"${SNOWFLAKE_USER}\"" >> $config_file
+                echo "role = \"${SNOWFLAKE_SCHEMACHANGE_ROLE}\"" >> $config_file
+                echo "password = \"${SNOWFLAKE_PASSWORD}\"" >> $config_file
+                echo "warehouse = \"${SNOWFLAKE_WAREHOUSE}\"" >> $config_file
+                echo 'authenticator = "SNOWFLAKE"' >> $config_file
+
+                # Write config paths to environment
+                echo "SNOWFLAKE_CONFIG_PATH=$config_file" >> $GITHUB_ENV
+
+            - name: Install Snowflake CLI with DATA_ENGINEER config
+              uses: Snowflake-Labs/snowflake-cli-action@v1.5
+              with:
+                default-config-file-path: ${{ env.SNOWFLAKE_CONFIG_PATH }}
+
+            - name: Verify Snowflake CLI installation and connections
+              run: |
+                snow --version
+                snow connection test -c admin
+                snow connection test -c dpe
+
+            - name: Sanitize Clone Name
+              run: |
+                SNOWFLAKE_DOMAIN_CLONE_SANITIZED="${SNOWFLAKE_DOMAIN_CLONE//[^a-zA-Z0-9_]/_}"
+                echo "Clone name has been updated! The clone name will be: ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}"
+                echo "SNOWFLAKE_DOMAIN_CLONE_SANITIZED=${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}" >> $GITHUB_ENV
+
+            - name: Zero-copy clone the database
+              shell: bash
+              run: |
+                snow sql -q "CREATE OR REPLACE DATABASE $SNOWFLAKE_DOMAIN_CLONE_SANITIZED CLONE $SNOWFLAKE_DOMAIN_ORIGINAL;"
+
+            - name: Grant permissions to DATA_ENGINEER on cloned database
+              shell: bash
+              run: |
+                snow connection set-default admin
+
+                # Transfer ownership of: *ALL_ADMIN aggregate database roles (for each namespace/schema)
+                snow sql -q "GRANT OWNERSHIP ON DATABASE ROLE ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}.SYNAPSE_ALL_ADMIN TO ROLE DATA_ENGINEER;"
+                snow sql -q "GRANT OWNERSHIP ON DATABASE ROLE ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}.SYNAPSE_RAW_ALL_ADMIN TO ROLE DATA_ENGINEER;"
+                snow sql -q "GRANT OWNERSHIP ON DATABASE ROLE ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}.SCHEMACHANGE_ALL_ADMIN TO ROLE DATA_ENGINEER;"
+
+                # Transfer ownership of: ownership proxy database role
+                snow sql -q "GRANT OWNERSHIP ON DATABASE ROLE ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}.??? TO ROLE DATA_ENGINEER;"
```

Comment:
@philerooski I think this is a true blocker since the proxy role is necessary to have ownership of objects across schemas, right? Was there a ticket made for creating O2 in this [design doc](https://sagebionetworks.jira.com/wiki/spaces/DPE/pages/3882582041/Inter-namespace+privileges)?

I'm happy to make the ticket, and LMK what else I can do to move this forward

Reply:
Yes, I will need to implement O2 before we can finish the CI/clone work here. That's in the acceptance criteria of [the ticket](https://sagebionetworks.jira.com/browse/SNOW-218) so it will be done this sprint.

Reply:
Sounds good, no worries, I'll update my ticket accordingly